BraveRewardsUI/Wallet/Notifications/RewardsNotificationViewBuilder.swift

```diff
@@ -0,0 +1,111 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import BraveRewards
+
+struct RewardsNotificationViewBuilder {
+  
+  static func get(notification: RewardsNotification) -> WalletNotificationView? {
+    switch notification.kind {
+    case .autoContribute, .tipsProcessed, .verifiedPublisher:
+      return RewardsNotificationViewBuilder.get(messageNotification: notification)
+    case .grant, .grantAds, .insufficientFunds, .pendingNotEnoughFunds, .adsLaunch:
+      return RewardsNotificationViewBuilder.get(actionNotification: notification)
+    default:
+      return nil
+    }
+  }
+  
+  private static func get(actionNotification: RewardsNotification) -> WalletNotificationView? {
+    let category: WalletActionNotification.Category
+    let body: String
+    
+    switch actionNotification.kind {
+    case .grant:
+      body = Strings.NotificationGrantNotification
+      category = .grant
+    case .grantAds:
+      body = Strings.NotificationEarningsClaimDefault
+      category = .grant
+    case .insufficientFunds:
+      body = Strings.NotificationInsufficientFunds
+      category = .insufficientFunds
+    case .pendingNotEnoughFunds:
+      body = Strings.NotificationPendingNotEnoughFunds
+      category = .insufficientFunds
+    default:
+      assertionFailure("Undefined case for action notification")
+      return nil
+    }
+    let date = Date(timeIntervalSince1970: actionNotification.dateAdded)
+    return WalletActionNotificationView(
+      notification: WalletActionNotification(
+        category: category,
+        body: body,
+        date: date))
+  }
+  
+  private static func get(messageNotification: RewardsNotification) -> WalletNotificationView? {
+    var category: WalletMessageNotification.Category
+    let body: String
+    var alertType: WalletAlertNotification.Category?
+    switch messageNotification.kind {
+    case .autoContribute:
+      if let result = messageNotification.userInfo["result"] as? Int, let amount = messageNotification.userInfo["amount"] as? String {
```

Comment:
Use `BATResult` enum switch instead of magic numbers below by creating a `BATResult` from this line